### ISSUE TITLE
Missing nc_inq_filter_avail HDF4 1.10.7 & netcdf 4.8.1

### ISSUE THREAD

**Issue by snowman2** [Unidata/netcdf4-python#1209]
netCDF4-python==1.6.1 
```
     reading from setup.cfg...
      using nc-config...
      checking /usr/include ...
      hdf5 headers not found in /usr/include
      checking /usr/include/hdf5/serial ...
      HDF5 library version: 1.10.7 headers found in /usr/include/hdf5/serial
      using netcdf library version b'4.8.1'
      using Cython to compile netCDF4.pyx...
      netcdf lib has group rename capability
      netcdf lib has nc_inq_path function
      netcdf lib has nc_inq_format_extended function
      netcdf lib has nc_open_mem function
      netcdf lib has nc_create_mem function
      netcdf lib has cdf-5 format capability
      netcdf lib does not have netcdf4 parallel functions
      netcdf lib does not have pnetcdf parallel functions
      netcdf lib does not have bit-grooming/quantization functions
      netcdf lib does not have zstandard compression functions
      netcdf lib does not have bzip2 compression functions
      netcdf lib does not have blosc compression functions
      netcdf lib has szip compression functions
      netcdf lib does not have nc_set_alignment function
      NETCDF_PLUGIN_DIR not set, no netcdf compression plugins installed
      /opt/venv/lib/python3.9/site-packages/setuptools/installer.py:27: SetuptoolsDeprecationWarning: setuptools.installer is deprecated. Requirements should be satisfied by a PEP 517 installer.
        warnings.warn(
      running install
      /opt/venv/lib/python3.9/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
        warnings.warn(
      running build
      running build_py
      creating build
      creating build/lib.linux-x86_64-cpython-39
      creating build/lib.linux-x86_64-cpython-39/netCDF4
      copying src/netCDF4/__init__.py -> build/lib.linux-x86_64-cpython-39/netCDF4
      copying src/netCDF4/utils.py -> build/lib.linux-x86_64-cpython-39/netCDF4
      running build_ext
      cythoning src/netCDF4/_netCDF4.pyx to src/netCDF4/_netCDF4.c
      
      Error compiling Cython file:
      ------------------------------------------------------------
      ...
      **`has_szip_filter(self)`**
      returns True if szip compression filter is available"""
              cdef int ierr
              IF HAS_SZIP_SUPPORT:
                  with nogil:
                      ierr = nc_inq_filter_avail(self._grpid, H5Z_FILTER_SZIP)
```

**Comment by jswhit** [comment]
thanks for reporting this @snowman2.  Looks like we should not be using nc_inq_filter_avail for netcdf-c < 4.9.0.

**Comment by jswhit** [comment]
closed by PR #1210.  Should probably cut a new release for this.

**Comment by snowman2** [comment]
Thanks @jswhit :+1:
